### PR TITLE
 feat(contact): sharded the contact library

### PR DESCRIPTION
**Pull request by Nolan-Arnold**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently, the contact package is one monolithic import. 

Fixes: N/A

## What is the new behavior?
This splits it apart to reduce the total downloaded code.

## Does this PR introduce a breaking change?
```
[X] Yes
[ ] No
```

Many imports have moved.

## Other information

### CODE REVIEW

File: libs/contact/state/src/facades/contact.facade.spec.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import { cold } from 'jasmine-marbles';
+
+import {
+	reducer,
+	DaffContactSuccessSubmit,
+	DaffContactSubmit,
+	DaffContactFailedSubmit,
+} from '@daffodil/contact/state';
+
+import { DaffContactFeatureState } from '../selectors/contact.selector';
```
Comment (griest024) [inline libs/contact/state/src/facades/contact.facade.spec.ts:13]: Should be a `@daffodil` import.

File: libs/contact/tsconfig.json [hunk 2]
```diff
@@ -23,9 +23,21 @@
       "@daffodil/contact": [
```
Comment (griest024) [inline libs/contact/tsconfig.json:23]: The `dist` paths can all be replaced with a wildcard path alias.

**Review by griest024 — CHANGES_REQUESTED** [review]
[Selector spec reducer imports](https://github.com/graycoreio/daffodil/blob/7a15703c4f187d2462c0ca2ad176b4af43481795/libs/contact/state/src/selectors/contact.selector.spec.ts#L2) should be from `@daffodil`.

File: libs/contact/state/src/contact.module.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { NgModule } from '@angular/core';
+
+@NgModule({
+	declarations: [],
+	imports: [],
+	providers: [],
+})
+export class DaffContactModule {}
```
Comment (lderrickable) [inline libs/contact/state/src/contact.module.ts:8]: This should export/inject the effects and reducers, and should be called `DaffContactStateModule`

File: libs/contact/state/testing/src/public_api.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export { MockDaffContactFacade } from './mock-contact-facade';
+export { DaffContactTestingModule } from './contact-testing.module';
```
Comment (lderrickable) [inline libs/contact/state/testing/src/public_api.ts:2]: Should be `DaffContactStateTestingModule`

**Review by lderrickable — CHANGES_REQUESTED** [review]

File: libs/contact/state/src/facades/contact.facade.spec.ts (hunk 1 above)
Comment (Nolan-Arnold) [reply libs/contact/state/src/facades/contact.facade.spec.ts]: Fixed in 061de9d8ecfc2da4c3df5910d597a21954e70e76

File: libs/contact/tsconfig.json (hunk 2 above)
Comment (Nolan-Arnold) [reply libs/contact/tsconfig.json:8]: Fixed in 061de9d8ecfc2da4c3df5910d597a21954e70e76

File: libs/contact/state/src/contact.module.ts (hunk 3 above)
Comment (Nolan-Arnold) [reply libs/contact/state/src/contact.module.ts:12]: Fixed in 061de9d8ecfc2da4c3df5910d597a21954e70e76

File: libs/contact/state/testing/src/public_api.ts (hunk 4 above)
Comment (Nolan-Arnold) [reply libs/contact/state/testing/src/public_api.ts]: Fixed in 061de9d8ecfc2da4c3df5910d597a21954e70e76

**Review by griest024 — CHANGES_REQUESTED** [review]
Selector spec reducer imports should be from @daffodil.

**Comment by Nolan-Arnold** [comment]
> Selector spec reducer imports should be from @daffodil.

bb204e259d521d2463121b650fffd51b1b35d36b okay changes should be in here and I think that's all spec import paths fixed.

**Review by griest024 — APPROVED** [review]

File: libs/contact/state/src/contact.module.ts (hunk 3 above)
Comment (lderrickable) [reply libs/contact/state/src/contact.module.ts:12]: The name on the module should be changed to `DaffContactStateModule`. And the effect should be provided here unless there's a reason not to

Comment (Nolan-Arnold) [reply libs/contact/state/src/contact.module.ts]: Sorry for missing this comment awhile ago. Fixed in 3accd4e2f658fa0ce8269bc5fbf7e5367e610e65.

**Review by lderrickable — DISMISSED** [review]

**Comment by lderrickable** [comment]
This PR has unsigned commits. @Nolan-Arnold I think you need to do this to get your commits signed: https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/about-commit-signature-verification

**Comment by Nolan-Arnold** [comment]
> This PR has unsigned commits. @Nolan-Arnold I think you need to do this to get your commits signed: https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/about-commit-signature-verification

**Review by lderrickable — APPROVED** [review]

File: libs/contact/tsconfig.json [hunk 5]
```diff
@@ -2,30 +2,27 @@
...
-        "dist/driver/hubspot"
+      "@daffodil/contact/driver/in-memory": [
+        "libs/contact/driver/in-memory/src"
       ],
-      "@daffodil/contact": [
-        "libs/contact/src"
+      "@daffodil/contact/driver/hubspot": [
+        "libs/contact/driver/hubspot/src"
       ],
       "@daffodil/contact/testing": [
```
Comment (damienwebdev) [inline libs/contact/tsconfig.json:20]: This package name is wrong. It also looks like you're missing state/testing?

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/contact/tsconfig.json (hunk 5 above)
Comment (lderrickable) [reply libs/contact/tsconfig.json:20]: There's also a directory that needs deleted: `contact/testing/**`

Comment (Nolan-Arnold) [reply libs/contact/tsconfig.json]: Okay both fixed in 6deedfcb77d603ec3c6183bb04c499ef8a539067.

**Review by lderrickable — APPROVED** [review]

**Review by lderrickable — APPROVED** [review]

**Review by damienwebdev — APPROVED** [review]

**Pull request merged**